### PR TITLE
fix: fix and test array function if has nil

### PR DESCRIPTION
**Pull request by Yisaer**

### CODE REVIEW

File: internal/binder/function/funcs_array.go [hunk 1]
```diff
@@ -155,30 +153,7 @@ func registerArrayFunc() {
 		val: func(ctx api.FunctionContext, args []ast.Expr) error {
```
Comment (ngjaying) [inline internal/binder/function/funcs_array.go:153]: If array is nil, should just return nil instead of error?

**Review by ngjaying — CHANGES_REQUESTED** [review]
The array functions should support nil array instead of just throw out an error. Please go through all functions and check.

File: internal/binder/function/funcs_array.go [hunk 2]
```diff
@@ -155,30 +153,7 @@ func registerArrayFunc() {
...
 			return ValidateLen(2, len(args))
 		},
-		check: returnNilIfHasAnyNil,
-	}
-	builtins["array_position"] = builtinFunc{
```
[the remark this answers is not in the document]
Comment (Yisaer) [reply internal/binder/function/funcs_array.go:160]: it's duplicated.

File: docs/en_US/guide/sources/plugin/sql.md [hunk 3]
```diff
@@ -107,13 +107,12 @@ The target database url
 * `indexFieldType`: column type for the indexField, if it is dateTime type, must set this field with `DATETIME`
 * `dateTimeFormat`: data time format for the index field
 
-::: v-pre
+
```
Comment (ngjaying) [inline docs/en_US/guide/sources/plugin/sql.md:110]: Don't remove this

File: docs/zh_CN/sqls/functions/array_functions.md [hunk 4]
```diff
@@ -139,23 +139,23 @@ array_flatten(array)
...
+例如，传入参数为 [[1, 4], [2, 3]]，则返回 [1, 4, 2, 3]。array 为 nil 时则固定返回 0。
 
 ## ARRAY_DISTINCT
 
 ```text
 array_distinct(array)
 ```
 
-返回一个去重的数组，即将数组中的重复元素去除。
+返回一个去重的数组，即将数组中的重复元素去除。array 为 nil 时则固定返回 0。
```
Comment (ngjaying) [inline docs/zh_CN/sqls/functions/array_functions.md:150]: return nil

File: docs/zh_CN/sqls/functions/array_functions.md [hunk 5]
```diff
@@ -139,23 +139,23 @@ array_flatten(array)
...
 
 ## ARRAY_MAP
 
 ```text
 array_map(function_name, array)
 ```
 
-返回一个新的数组，其中包含对给定数组中的每个元素应用给定函数的结果。
+返回一个新的数组，其中包含对给定数组中的每个元素应用给定函数的结果。array 为 nil 时则固定返回 0。
```
Comment (ngjaying) [inline docs/zh_CN/sqls/functions/array_functions.md:158]: return nil

**Review by ngjaying — CHANGES_REQUESTED** [review]
array_concat doc is not updated

File: docs/en_US/guide/sinks/overview.md [hunk 6]
```diff
@@ -86,10 +86,10 @@ Each action can define its own properties. There are several common properties:
...
-| lingerInterval       | int  0                               | Specify the interval time for buffer messages before seding, the unit is millisecond. The sink will block sending messages until the buffer sending interval reaches this value. lingerInterval can be used together with batchSize to trigger sending when any condition is met.                                                                                                                                                                                                                                                                                                                                                                          |                                    |
+| resendIndicatorField | string: default to global definition | field name of the resend cache, the field type must be a bool value. If the field is set, it will be set to true when resending. e.g., if resendIndicatorField is `resend`, then the `resend` field will be set to true when resending the cache. |
```
Comment (ngjaying) [inline docs/en_US/guide/sinks/overview.md:89]: why change this file, it is irrelevant. The sme to sql.md

**Review by ngjaying — CHANGES_REQUESTED** [review]

**Review by ngjaying — APPROVED** [review]

**Pull request merged**